analyze_functional_test_results.py

```diff
@@ -76,17 +96,24 @@ def percent_matched_routes(
         if person_id in golden_response_set and person_id in responses_to_validate:
             golden_entry = golden_response_set[person_id]
             to_validate_entry = responses_to_validate[person_id]
+            golden_compare = copy.deepcopy(golden_entry)
```

Comment:
you could just do dict comprehension here to make it cleaner?

golden_compare = {k: v for k, v in golden_entry.items() if k != "query_time"}
then you don't need to pop 🤷 

but you might need to copy/deep copy the values.
does it matter that we modify the distances elsewhere?

Reply:
Good call, this does seem a bit cleaner than the copy + pop approach I'm taking now. re: distances being modified, I don't _really_ think it matters? It felt a lot safer to do the copy and to be 100% sure (I was imagining potential future checks around route distance or something), but considering we're only doing simple rounding on the distance, I'm not sure we should really care.

I'll switch to using dict comp here, and I'm going to not copy/deepcopy the values, but will add a comment that distances will be modified here in case we ever revisit this